openedx/adg/lms/conftest.py

```diff
@@ -47,3 +53,30 @@ def course_overviews(current_time):
         'test_course1': CourseOverviewFactory(language='en', **course_start_end_date),
         'test_course2': CourseOverviewFactory(language='ar', **course_start_end_date),
     }
+
+
+@pytest.fixture(name='prereq_course_groups')
+def create_prerequisite_course_group(request):
+    """
+    This fixture takes in a tuple to set the batch size of course groups to be created and to make these
+    course groups either program or business line prerequisite e.g. (10, 'business_line_pre_req')
+    """
+
+    if request.param[1] == BUSINESS_LINE_PRE_REQ:
+        course_groups = MultilingualCourseGroupFactory.create_batch(
+            request.param[0], is_program_prerequisite=False, is_common_business_line_prerequisite=True
+        )
+    else:
+        course_groups = MultilingualCourseGroupFactory.create_batch(request.param[0])
+
+    for course_group in course_groups:
+        now = timezone.now()
+        # Open prereq courses
+        MultilingualCourseFactory.create_batch(
+            2, course__start_date=now, course__end_date=now + timedelta(days=1), multilingual_course_group=course_group
+        )
+        # Ended prereq course
+        MultilingualCourseFactory(
+            course__start_date=now + timedelta(days=2), course__end_date=now, multilingual_course_group=course_group
+        )
```

Comment:
Is such a course valid, whose end date is less than start date? From the CMS side, it doesn't allow you to create such a course. If we want to create an ended course, then all we have to do is make sure its end date is in the past. Even in that case, the start date would be less than end date. 

When creating test data, its best to mimic real life, valid data.